tests/installation/disk_activation.pm

```diff
@@ -26,79 +33,91 @@ sub run {
-        elsif (get_var('FORMAT_DASD_YAST')) {
-            send_key 'alt-s';                           # select all
-            assert_screen 'dasd-selected';
-            send_key 'alt-a';                           # perform action button
-            if (check_screen 'dasd-device-formatted') {
+            # we need to type backspace to delete the content of the input field in textmode
+            if (check_var("VIDEOMODE", "text")) {
+                send_key 'alt-m';    # minimum channel ID
+                for (1 .. 9) { send_key "backspace"; }
+                type_string '0.0.0150';
+                send_key 'alt-x';    # maximum channel ID
+                for (1 .. 9) { send_key "backspace"; }
+                type_string '0.0.0150';
+                send_key 'alt-f';    # filter button
+                assert_screen 'dasd-unselected';
+                send_key 'alt-s';    # select all
+                assert_screen 'dasd-selected';
+                send_key 'alt-a';    # perform action button
+                assert_screen 'action-list';
+                send_key 'alt-a';    # activate
+            }
+            else {
+                send_key 'alt-m';    # minimum channel ID
+                type_string '0.0.0150';
+                send_key 'alt-x';    # maximum channel ID
+                type_string '0.0.0150';
+                send_key 'alt-f';    # filter button
+                assert_screen 'dasd-unselected';
+                send_key 'alt-s';    # select all
+                assert_screen 'dasd-selected';
+                send_key 'alt-a';    # perform action button
                 assert_screen 'action-list';
-                send_key 'f';
-                send_key 'f';                           # Pressing f twice because of bsc#940817
-                send_key 'ret';
-                assert_screen 'confirm-dasd-format';    # confirmation popup
-                send_key 'alt-y';
+                send_key 'a';        # activate
+            }
+
+            # sometimes it happens, that the DASD is in a unstable state, so
+            # if the systems wants to format the DASD by itself, do it.
+            if (check_screen 'dasd-format-device', 10) {    # format device pop-up
+                send_key 'alt-o';                           # continue
                 format_dasd;
             }
+
+            # format DASD if the variable is that, because we format it usually pre-installation
+            elsif (get_var('FORMAT_DASD_YAST')) {
+                send_key 'alt-s';                           # select all
+                assert_screen 'dasd-selected';
+                send_key 'alt-a';                           # perform action button
+                if (check_screen 'dasd-device-formatted') {
+                    assert_screen 'action-list';
+                    send_key 'f';
+                    send_key 'f';                           # Pressing f twice because of bsc#940817
```

Comment:
Because of issues with key presses in long term this may fail, if one of key presses is not recognized. Is there anything we could use to verify that it happened? As it's a bug, would be nice to introduce soft failure with reference. Maybe send_key_until_needle_matches can fit our needs.

Reply:
also, pretty old code which was working perfectly fine for a long time,
only appears as new due to the new intdentation